src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```
@@ -10,42 +15,191 @@ namespace Infrastructure.Common
+                // if it has been set to 'true' or 'false'.  If not set, fall into detection logic.
+                // This convention permits a developer to override any detection logic to
+                // enable or disable manually.
+                string environmentVariable = Environment.GetEnvironmentVariable(conditionName);
+                bool result = false;
+                if (!String.IsNullOrWhiteSpace(environmentVariable) && bool.TryParse(environmentVariable, out result))
+                {
+                    return result;
+                }
+                
+                // No enviromnent variable, so detect at runtime or default to 'false'
+                return detectFunc == null ? false : detectFunc();
+            });
+        }
 
-            return false;
+        // Returns the 'true' or 'false' for a condition whose name is
+        // contained in TestProperties.   If the property's value is
+        // 'true' or 'false', it will be used as is.  If the value is
+        // empty, and 'detectFunc' is specified, it will be invoked to
+        // evaluate the condition at runtime.  The value is cached, so
+        // 'detectFunc' will be called only once.
+        private static bool GetConditionValueFromTestProperties(string conditionName, Func<bool> detectFunc = null)
+        {
+            string propertyValue;
+            return GetConditionValue(conditionName, () =>
+            {
+                propertyValue = TestProperties.GetProperty(conditionName);
+                if (string.IsNullOrWhiteSpace(propertyValue))
+                {
+                    // Empty string means auto-detect if possible, else false
+                    return (detectFunc == null) ? false : detectFunc();
+                }
+                else
+                {
+                    bool conditionResult = false;
+                    if (!bool.TryParse(propertyValue, out conditionResult))
```

If we're looking for consistent results and error discovery, rather than being tolerant of bad input, we should hard throw instead of ignoring or guessing - both "guessing" and "ignoring" behaviours are strictly wrong.